metricbeat/mb/registry.go

```diff
@@ -222,16 +236,22 @@ func (r *Register) metricSetRegistration(module, name string) (MetricSetRegistra
 	name = strings.ToLower(name)
 
 	metricSets, exists := r.metricSets[module]
-	if !exists {
-		return MetricSetRegistration{}, fmt.Errorf("metricset '%s/%s' is not registered, module not found", module, name)
+	if exists {
+		registration, exists := metricSets[name]
+		if exists {
+			return registration, nil
+		}
 	}
 
-	registration, exists := metricSets[name]
-	if !exists {
-		return MetricSetRegistration{}, fmt.Errorf("metricset '%s/%s' is not registered, metricset not found", module, name)
+	if source := r.secondarySource; source != nil && source.HasMetricSet(module, name) {
+		registration, err := source.MetricSetRegistration(r, module, name)
+		if err != nil {
+			return MetricSetRegistration{}, errors.Wrapf(err, "failed to obtain registration for non-registered metricset '%s/%s'", module, name)
+		}
+		return registration, nil
 	}
 
-	return registration, nil
+	return MetricSetRegistration{}, fmt.Errorf("metricset '%s/%s' not found", module, name)
```

Comment:
this message is going to be helpful :100:

Reply:
This was already there, I have only moved it, and removed the mention to registration :slightly_smiling_face: